unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/Scene/Messaging/MessagingBus.cs

```diff
@@ -73,79 +74,69 @@ public void Stop()
 
         public void Enqueue(QueuedSceneMessage message, QueueMode queueMode = QueueMode.Reliable)
         {
-            bool enqueued = true;
-
-            // When removing an entity we have to ensure that the enqueued lossy messages after it are processed and not replaced
-            if (message is QueuedSceneMessage_Scene queuedSceneMessage && queuedSceneMessage.payload is Protocol.RemoveEntity removeEntityPayload)
+            lock (unreliableMessages)
```

Comment:
IIRC this methods were not correctly covered by tests, so we should revise this very very carefully so the exact same logic is preserved and/or add test coverage. As adding tests would be way painful for this class, maybe we can try to do a very very careful review and add the tests as technical debt. 

I'll do my best from here but the diff layout seems to get crazy due to code being moved around.

Reply:
I think that in the end, trying to make the enqueuing multithreaded without ensuring its correctly covered by tests first was a misstep on our part.

Reply:
Yes, this was totally an oversight on my part, also those tests are explicit and they are not running on the CI, gotta fix them and add more coverage